bbbc021_feature_reduction_tests.py

```diff
@@ -0,0 +1,247 @@
+from sklearn.manifold import TSNE
+import umap
+
+from microscopium import io
+from microscopium.preprocess import montage_stream
+from microscopium.preprocess import correct_multiimage_illumination
+from microscopium.preprocess import find_background_illumination
+from microscopium.features import default_feature_map
+
+
+#FILE_NAME_PREFIX = "Week1_150607_"
+IMAGE_FILE_PATH = "/data/BBBC021/data.broadinstitute.org/bbbc/BBBC021/"
+OUTPUT_FILE_PATH = "/data/bbbc_out/"
+
+FEATURES_FILE = "./all_features.csv"
+DATA_FILE = OUTPUT_FILE_PATH + "Data.csv"
+DATA_TSNE = OUTPUT_FILE_PATH + "Data_TSNE.csv"
+DATA_UMAP = OUTPUT_FILE_PATH + "Data_UMAP.csv"
+
+def main():
+
+    ## get valid filenames and build output filenames
+    filenames, names_illum = get_valid_file_names(IMAGE_FILE_PATH) 
+
+    ## run illumination correction on images
+   # for i, directory_files in enumerate(filenames):
+   #     run_illum(directory_files, names_illum[i])
+   #     print("Directory processed: ", directory_files[0])
+   
+    all_names_montage = []
+    ## montage illumed images
+    for i, filename_dir in enumerate(names_illum): 
+         sorted_filenames = sorted(filename_dir)
+         names_montage = make_montage_names(sorted_filenames)
+         all_names_montage.extend(names_montage)
+   #      run_montage(sorted_filenames, names_montage)
+   #      print("Directory {} out of {} processed".format(i, len(names_illum)))
+
+
+     ## run features on images
+    ims = map(io.imread, all_names_montage)
+    output_features(ims, all_names_montage, FEATURES_FILE)
+    
+     ## get x y coordinates
+    coords_pca = pca_transform(FEATURES_FILE)
+    print("Coords PCA retrieved")
+     ## generate CSV of coordinates
+    generate_bokeh_csv(coords_pca, all_names_montage, DATA_FILE)
+
+    coords_tsne = tsne_transform(FEATURES_FILE)
+    print("Coords tsne retrieved")
+    generate_bokeh_csv(coords_tsne, all_names_montage, DATA_TSNE)
+
+    coords_umap = umap_transform(FEATURES_FILE)
+    print("Coords umap retrieved")
+    generate_bokeh_csv(coords_umap, all_names_montage, DATA_UMAP)
+
+def make_montage_names(filenames):
+    """
+    Construct filenames for the montaged images
+
+    :param filenames: names of the illumed images with separate quadrants and channels
+    :return names_montage: names for each file which will result from montaging
+    """
+    filename_reg = r'(.*Week._.*_)(...)(_s._w..*)(_illum)(\.tif)$'
+    names_montage = []
+    for filename in filenames:
+        match = re.search(filename_reg, filename)
+        if match:
+            potential_name = match.group(1) + match.group(2) + "_montaged" + match.group(5)
+            if potential_name not in names_montage:
+                names_montage.append(potential_name)
+    return names_montage
+
+def get_valid_file_names(filepath):
+    """
+    Get full filenames relative to top level directory for each file in the BBBC trial, and
+    construct filenames with paths for saving output
```

Comment:
Please read and conform to PEP 257. Specifically, first docstring line should be inline with the triple quotes and fit in 80c. You can expand more after a blank line.

Reply:
Link: https://www.python.org/dev/peps/pep-0257/